src/api/java/mezz/jei/api/gui/handlers/IGuiClickableArea.java

```diff
@@ -18,6 +18,16 @@
 	 */
 	Rectangle2d getArea();
 
+	/**
+	 * Used to check if the mouse is currently over the hover/click area for this {@link IGuiClickableArea}.
+	 * When hovered, the message from {@link #getTooltipStrings()} will be displayed.
+	 * When clicked, {@link #onClick(IFocusFactory, IRecipesGui)} will be called.
+	 */
```

Comment:
last thing: when adding something to the API, we must bump the minor version of JEI, and add a comment like `@since JEI version 1.2.3`. That way users of the API know they need to require at least this version as a dependency if they use the feature.